src/bacnet/basic/object/device.c

```diff
@@ -844,6 +845,20 @@ bool Device_Set_Model_Name(const char *name, size_t length)
     return status;
 }
 
+bool Device_Set_Serial_Number(const char *serial_number, size_t length)
+{
+    bool status = false; /*return value */
+
+    if (length < sizeof(Serial_Number)) {
+        memmove(Serial_Number, serial_number, length);
```

Comment:
strncpy instead of memmove?

Reply:
memmove is what is used in the other functions in this file, so I was just following its lead for compatibility reasons